src/commands/explorer/groupBy.ts

```diff
@@ -5,13 +5,23 @@
 
 import { IActionContext } from "@microsoft/vscode-azext-utils";
 import { ext } from "../../extensionVariables";
+import { getArmTagKeys } from "../../utils/azureUtils";
+import { localize } from "../../utils/localize";
 import { settingUtils } from "../../utils/settingUtils";
 
 export function buildGroupByCommand(setting: string) {
     return (context: IActionContext): Promise<void> => groupBy(context, setting);
 }
 
 async function groupBy(context: IActionContext, setting: string): Promise<void> {
+    if (setting === 'armTag') {
+        const armTagKeys = await getArmTagKeys(context);
+        const tag = await context.ui.showQuickPick(Array.from(armTagKeys).map(key => { return { label: key } }), {
+            placeHolder: localize('groupByArmTagKey', 'Select the tag key to group by...')
+        });
+        setting += `-${tag.label}`;
```

Comment:
Appending it to the groupBy setting string like this makes me a bit uneasy but I understand why we need to do it this way so it's ok.

Reply:
Yeah, I was originally messing with groupBy setting that is only `armTag` and then other setting that was the key that is being used.  However, that created a lot of if/else clauses to special case when the groupBy setting was armTag which kind of defeated the simplicity of the group configuration code.